File: src/NubArca.Api/Users/PasswordPolicy.cs
namespace NubArca.Api.Users;

// Shared minimum password policy for admin-set and self-service password
// changes. Deliberately simple: length bounds + not-all-whitespace. No
// character-class complexity rules — the project has none today and this
// slice does not introduce them.
public static class PasswordPolicy
{
    public const int MinLength = 10;
    public const int MaxLength = 256;

    public const string ErrorMessage = "Password does not meet the minimum requirements.";

    public static bool IsValid(string? password) => TryValidate(password, out _);

    public static bool TryValidate(string? password, out string? error)
    {
        if (string.IsNullOrEmpty(password)
            || password.Trim().Length == 0
            || password.Length < MinLength
            || password.Length > MaxLength)
        {
            error = ErrorMessage;
            return false;
        }

        error = null;
        return true;
    }
}

File: frontend/src/account/passwordPolicy.ts
// Mirrors NubArca.Api.Users.PasswordPolicy on the backend. The backend is
// the source of truth (this is just fast client-side feedback before a
// round-trip) — length bounds + not-all-whitespace, no complexity rules.
export const PasswordPolicy = {
  minLength: 10,
  maxLength: 256,

  isValid(password: string): boolean {
    const trimmed = password.trim();
    return (
      trimmed.length > 0
      && password.length >= PasswordPolicy.minLength
      && password.length <= PasswordPolicy.maxLength
    );
  },
};
